csrc/scheduler/hopper_multi_matmul.cpp

```diff
@@ -554,6 +562,14 @@ void HopperMultipleMatmulScheduler::scheduleEpilogue() {
       // NOTE: cacheBefore does not work with blockTileTensors
       TensorView* d_smem = cacheAfter(dc, LoadStoreOpType::Set);
 
+      std::vector<TensorView*> tvs_to_schedule{d, d_smem};
+      if (std::find(mma_results_.begin(), mma_results_.end(), dc) ==
+          mma_results_.end()) {
+        // Skip scheduling dc if it is an mma_result. This can happen if we are
+        // not casting back to half-precision in the output
+        tvs_to_schedule.push_back(dc);
+      }
+
```

Comment:
Are we still using `stmatrix` if the output is fp32 and there wasn't a cast?

Reply:
Yes, if you enable `use_smem_epilogue` in the included test we hit an error in `scheduleStMatrixForMmaOutput`.

Reply:
So we need an if/else here that checks the dtype of `d_smem` and schedules with vectorized stores instead if not 16bit https://github.com/NVIDIA/Fuser/blob/230f633eb995d6690ba67c57117c3b94ea26f25b/csrc/scheduler/hopper_multi_matmul.cpp#L584-L588